ingest/src/projectSources.ts

```diff
@@ -504,6 +506,53 @@ const practicalAggregationsDataSource = async () => {
   return await makeMdOnGithubDataSource(practicalAggregationsConfig);
 };
 
+export const terraformProviderSourceConstructor = async () => {
+  const siteBaseUrl =
+    "https://registry.terraform.io/providers/mongodb/mongodbatlas/latest/docs";
+  return await makeGitDataSource({
+    name: "atlas-terraform-provider",
+    repoUri: "https://github.com/mongodb/terraform-provider-mongodbatlas.git",
+    repoOptions: {
+      "--depth": 1,
+      "--branch": "master",
+    },
+    metadata: {
+      productName: "mongodbatlas Terraform Provider",
+      tags: ["docs", "terraform", "atlas", "hcl"],
+    },
+    filter: (path: string) =>
+      path.includes("website/docs") && path.endsWith(".markdown"),
+    handlePage: async function (path, content) {
+      const { metadata, body } = extractFrontMatter<{ page_title: string }>(
+        content
+      );
+      let url: string;
+      if (path.includes("website/docs/d/")) {
+        url =
+          siteBaseUrl +
+          path
+            .replace("website/docs/d", "data-sources")
+            .replace(".markdown", "");
+      } else if (path.includes("website/docs/r/")) {
+        url =
+          siteBaseUrl +
+          path.replace("website/docs/r", "resources").replace(".markdown", "");
+      } else {
+        url =
+          siteBaseUrl +
+          path.replace("website/docs/", "").replace(".markdown", "");
+      }
```

Comment:
The construction 
```
let temporarilyUndefinedVariable;
if (...) {
  temporarilyUndefinedVariable = thiz
} else if (...) {
  temporarilyUndefinedVariable = that  
} ...
```

is a pretty good indicator for refactor.

```js
const url = getUrlForBlahblah(path);
...
function getUrlForBlahblah(url) {
  if (...) {
    return thiz
  } else if (...) { ...
    return that
  } ...
}
```

This will actually also highlight dependency sprawl (think: opportunity for merging into a single findContentFunc style thing?) as it forces you to move scope variables into a parameter list for the helper function.

Btw, I would not export or explicitly test the helper function (`getUrlForBlahblah()` in this example).